examples/hpc/dla-driver/src/lib.rs

```diff
@@ -199,10 +199,10 @@ impl Dla {
         }
     }
 
-    pub fn write_data_bank(&self, offset: usize, buf: &mut [u8]) {
+    pub fn write_data_bank(&self, offset: usize, buf: &mut [i8]) {
         //sprintln!("\nWrite to bank {:#x}, data: {:?}", offset, buf);
         for (i, b) in buf.iter().enumerate() {
-            unsafe { ptr::write_volatile((MEMORY_BANK_BASE_ADDR + offset + i) as *mut u8, *b) };
+            unsafe { ptr::write_volatile((MEMORY_BANK_BASE_ADDR + offset + i) as *mut i8, *b) };
```

Comment:
You might want to elide the type in these assignments, if types are obvious from context, e.g. `write_volatile(ADDR as *mut _, value)`. Say, you have a function like `write_u8()`, there is no need to re-specify the type in the implementation. This might come down to preference at the end of the day.

Reply:
Removed explicit type assignments.